src/modules/mavlink/streams/GPS_RAW_INT.hpp

```diff
@@ -100,8 +104,19 @@ class MavlinkStreamGPSRawInt : public MavlinkStream
 			}
 
 			mavlink_msg_gps_raw_int_send_struct(_mavlink->get_channel(), &msg);
+			_last_send_ts = gps.timestamp;
 
 			return true;
+
+		} else if ((now = hrt_absolute_time()) > _last_send_ts + _no_gps_send_interval) {
+			msg.fix_type = GPS_FIX_TYPE_NO_GPS;
```

Comment:
This is kind of pedantic, but I think we should only be doing this if GPS was ever present.

If the `_sensor_gps_sub.update()` fails you could simply do a `_sensor_gps_sub.copy(&gps)` and check gps.timestamp !=0.

Reply:
That's a good point. 👍  I assumed that GPS would always be present, but not necessarily. I will look into it

Reply:
I have not used `_sensor_gps_sub.copy(&gps)` in the end, because it returns false if there are no publishers. If I killed the gps process with gps stop, it always returned false and the contents seemed invalid. I think this solution is quite robust.

@dagar What do you think?

Reply:
Looks good, thanks.